modules/aws-emr-cluster/main.tf

```
@@ -17,10 +17,10 @@ resource "aws_emr_cluster" "emr-cluster" {
   ec2_attributes {
     subnet_id                         = var.subnet_id
     emr_managed_master_security_group = var.emr_managed_master_sg_id
-    additional_master_security_groups = var.emr_additional_master_sg_id
+    additional_master_security_groups = join(", ", [for s in var.emr_managed_master_sg_ids : s])
     emr_managed_slave_security_group  = var.emr_managed_core_sg_id
-    additional_slave_security_groups  = var.emr_additional_core_sg_id
-    service_access_security_group     = var.emr_service_access_sg_id
+    additional_slave_security_groups  = join(", ", [for s in var.emr_managed_core_sg_ids : s])
+    service_access_security_group     = element(var.emr_service_access_sg_ids, 0)
```

I think we'll need to check if this list is empty.
```
The index is zero-based. This function produces an error if used with an empty list. The index must be a non-negative integer.
```
https://www.terraform.io/docs/language/functions/element.html

Maybe something like this?
```suggestion
    service_access_security_group     = length(var.emr_service_access_sg_ids) > 0 ? element(var.emr_service_access_sg_ids, 0) : null
```

Although I don't think it'll be valid to not specify it, but I think the error message might be better if we let amazon tell them rather than having the terraform not run.

---

Omg! I had this! I dont know why i deleted it! Will fix it

---

done